### PR TITLE
Convert hui-horizontal-stack-card to TypeScript/LitElement

### PR DESCRIPTION
**Pull request by iantrich**

### CODE REVIEW

File: src/panels/lovelace/cards/hui-stack-card.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        return html`
+      ${this.renderStyle()}
+      <div id="root">
+        ${this.config.cards.map((card) => this.createCardElement(card))}
+      </div>
+    `;
+    }
+
+    protected renderStyle() {
```
Comment (zsarnett) [inline src/panels/lovelace/cards/hui-stack-card.ts]: You don't need this function if you aren't using it

Comment (iantrich) [reply src/panels/lovelace/cards/hui-stack-card.ts]: It is used above on line 47 and is overridden by `HuiHorizontalStackCard` which extends this class

**Comment by balloob** [comment]
We should incorporate the fixes from #1856 into this PR.

File: src/panels/lovelace/cards/hui-stack-card.ts (hunk 1 above)
Comment (iantrich) [reply src/panels/lovelace/cards/hui-stack-card.ts]: Perfect! As @thomasloven [mentioned](https://github.com/home-assistant/home-assistant-polymer/pull/1853#issuecomment-432943050) might be a good idea for card with image backgrounds as well

File: src/panels/lovelace/cards/hui-stack-card.ts [hunk 2]
```diff
@@ -9,55 +10,58 @@ interface Config extends LovelaceConfig {
...
+      element.hass = this._hass;
     }
+  }
 
-    public setConfig(config: Config) {
-        if (!config || !config.cards || !Array.isArray(config.cards)) {
-            throw new Error("Card config incorrect");
-        }
-        this.config = config;
+  public getCardSize(): number {
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-stack-card.ts]: we should mark this abstract.

**Review by balloob — APPROVED** [review]
ok to merge when getCardSize is also marked as abstract.

**Pull request merged**